tests/lib/rules/class-methods-use-this.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @fileoverview Tests for class-methods-use-this rule.
+ * @author Patrick Williams
+ * @copyright 2016 Patrick Williams. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var rule = require("../../../lib/rules/class-methods-use-this");
+var RuleTester = require("../../../lib/testers/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester();
+ruleTester.run("class-methods-use-this", rule, {
+    valid: [
+        {code: "class A { constructor() {} }", parserOptions: { ecmaVersion: 6 }},
+        {code: "class A { foo() {this} }", parserOptions: { ecmaVersion: 6 }},
```

Comment:
just added this test, and the corresponding invalid version...which honestly is baffling to me.

Here's the invalid test:
`class A { foo() { () => undefined; } }`

This fails the way I would want it to, but not the way I *expected* it to. `=>` binds this, so at some point `this` is being used no? Anyhow, I'm happy with the additional tests, even if I don't fully understand them.